src/instance.rs

```diff
@@ -208,15 +225,17 @@ impl<'py, T> Bound<'py, T> {
 
     /// Removes the connection for this `Bound<T>` from the GIL, allowing
     /// it to cross thread boundaries.
+    #[inline]
     pub fn unbind(self) -> Py<T> {
         // Safety: the type T is known to be correct and the ownership of the
         // pointer is transferred to the new Py<T> instance.
-        unsafe { Py::from_non_null(self.into_non_null()) }
+        unsafe { Py::from_non_null(ManuallyDrop::new(self).1 .0) }
```

Comment:
This looks a bit weird. Maybe an intermediate binding or some parentheses could improve it?

Reply:
For now I'll go for

```rust
        let non_null = (ManuallyDrop::new(self).1).0;
        unsafe { Py::from_non_null(non_null) }
```